Client/Frontend/Browser/TabManager.swift

```diff
@@ -636,16 +634,22 @@ class TabManager: NSObject {
         guard let tempTabs = self.tempTabs, tempTabs.count > 0 else {
             return
         }
-        let tabsCopy = normalTabs
+
+        let tabsCopy = tabs(withType: .regular)
+
         restoreTabs(tempTabs)
         self.isRestoring = true
+
         for tab in tempTabs {
             tab.showContent(true)
         }
-        if !tempTabs[0].isPrivate {
+
+        let tab = tempTabs.first
+        if !TabType.of(tab).isPrivate {
```

Comment:
See this a decent amount. Do you think it is worth to include a facade for `isPrivate` directly on the tab object?

`!tab.isPrivate`

Or just too much bloat?

Reply:
I did think about this and discuss with @iccub but a `tab` in this and other instances can be `nil`, therefore `tab?.isPrivate` would go back to our original problem of have to provide a default tab type, i.e. `tab?.isPrivate ?? .regular`